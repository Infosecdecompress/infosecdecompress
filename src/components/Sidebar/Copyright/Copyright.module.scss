@use "sass:color";

@import "../../../assets/scss/variables";
@import "../../../assets/scss/mixins";

.copyright {
  color: color.adjust($color-gray, $whiteness: 18%);
  font-size: $typographic-small-font-size;
}
