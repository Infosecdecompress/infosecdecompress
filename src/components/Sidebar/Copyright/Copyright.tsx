import React from 'react';
import styles from './Copyright.module.scss';

type Props = {
  copyright: string
};

const Copyright: React.FC<Props> = ({ copyright }) => (
  <div className={styles['copyright']}>
    {copyright}
  </div>
);

export default Copyright;
