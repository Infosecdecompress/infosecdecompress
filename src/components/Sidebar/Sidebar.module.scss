@import "../../assets/scss/variables";
@import "../../assets/scss/mixins";

.sidebar {
  width: 100%;

  .inner {
    @include padding(1, 0.75, 0);
    position: relative;
  }
}

@include breakpoint-sm {
  .sidebar {
    lost-column: 5/12;

    .inner {
      @include padding(1.25, 0.75, 0);

      &:after {
        background: $color-gray-border;
        background: linear-gradient(
          to bottom,
          $color-gray-border 0%,
          $color-gray-border 48%,
          $color-white 100%
        );
        bottom: 0;
        content: "";
        height: 540px;
        position: absolute;
        right: -10px;
        top: 30px;
        width: 1px;
      }
    }
  }
}

@include breakpoint-md {
  .sidebar {
    lost-column: 1/3;

    .inner {
      @include padding-equal(1.5);
    }
  }
}
