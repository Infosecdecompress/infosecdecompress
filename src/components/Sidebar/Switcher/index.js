// @flow strict
export { default } from './Switcher';
