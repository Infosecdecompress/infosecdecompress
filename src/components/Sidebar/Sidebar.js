// @flow strict
import React from 'react';
import Author from './Author';
import Contacts from './Contacts';
import Copyright from './Copyright';
import Menu from './Menu';
import styles from './Sidebar.module.scss';
import { useSiteMetadata } from '../../hooks';
import Search from "../SearchContainer";
import Switcher from './Switcher';

const siteConfig = require('../../../config');

type Props = {
  isIndex?: boolean,
};

const Sidebar = ({ isIndex }: Props) => {
  const { author, copyright, menu } = useSiteMetadata();
  const { darkFunctionalityIsOn } = siteConfig;

  return (
    <div className={styles['sidebar']}>
      <div className={styles['sidebar__inner']}>
        <Author author={author} isIndex={isIndex} />
        <Menu menu={menu} />
        <Contacts contacts={author.contacts} />
        <Search />
        {darkFunctionalityIsOn ? <Switcher /> : null}
        <Copyright copyright={copyright} />
        </div>
      </div>
  );
};

export default Sidebar;
