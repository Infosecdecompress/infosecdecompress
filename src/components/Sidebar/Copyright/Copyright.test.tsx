import React from "react";
import renderer from "react-test-renderer";

import { Copyright } from "@/components/Sidebar/Copyright";
import * as mocks from "@/mocks";

describe("Copyright", () => {
  it("renders correctly", () => {
    const props = { copyright: mocks.siteMetadata.site.siteMetadata.copyright };
    const tree = renderer.create(<Copyright {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
