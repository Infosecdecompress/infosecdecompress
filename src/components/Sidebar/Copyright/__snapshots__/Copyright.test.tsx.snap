// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyright renders correctly 1`] = `
<div>
  Content is licensed under CC BY-SA 4.0
</div>
`;

exports[`Copyright renders correctly with empty string 1`] = `<div />`;
